alacritty/src/event.rs

```diff
@@ -844,10 +844,10 @@ impl<'a, N: Notify + 'a, T: EventListener> ActionContext<'a, N, T> {
         // Clear focused match.
         self.search_state.focused_match = None;
 
-        // The viewport reset logic is only needed for vi mode, since without it our origin is
-        // always at the current display offset instead of at the vi cursor position which we need
-        // to recover to.
-        if !self.terminal.mode().contains(TermMode::VI) {
+        // We don't need to reset a vi-mode cursor when at least one of vi-mode and search-mode
+        // isn't active because a vi-mode cursor is invisible, or even if not so, searching can't
+        // change both the cursor position and display offset.
```

Comment:
Why wouldn't searching be able to change both the cursor position and display offset? That's pretty much exactly what search does when in vi mode.

Reply:
It means incremental searching of search-mode doesn't happen, because search-mode isn't active.

What do you think if replaces the last part of above comment with below?

> Any vi-mode cursor position changing by incremental searching of search-mode, which we need to reset, doesn't happen.

Reply:
I think that this comment is basically impossible to understand.

Reply:
I can do it myself later.

Reply:
> I think that this comment is basically impossible to understand.

Really? Oh, that's sad.

> I can do it myself later.

OK, please do.

Reply:
Looking into this, I don't think this fix correctly encompasses the scope of this issue. Why the hell are we calling `cancel_search` when there isn't even a search active? We should never even get close to this conditional when search isn't even active.

It's probably best to either early return out of `cancel_search` or put it behind a conditional in `toggle_vi_mode`. While I haven't tested the full scope of these changes, I'd recommend you give that a try. It should probably work just as well?